docs/plugins/reindex.asciidoc

```diff
@@ -0,0 +1,608 @@
+
+Speaking of the `failures` element, it will contain a list of failures if
+there are any. The only "optional" sort of failures are version conflicts. All
+other failures always cause the operation to abort. While the first failure
+causes the abort, it's possible to encounter lots of subsequent failures due to
+batching so all failures are reported. Just in case you want them.
+
+It's important to remember that "abort" here doesn't mean "rollback". This
+operation isn't atomic. If it fails part way through then its changes have been
+made.
+
+Back to the API format, you can limit the copy to a single type. This will only
+update `tweet`s from the `twitter` index:
+
+[source,js]
+--------------------------------------------------
+POST /twitter/tweet/_update_by_query
+--------------------------------------------------
+// AUTOSENSE
+
+You can limit using the {ref}/query-dsl.html[Query DSL]. This will update
+all documents from the `twitter` index for the user `kimchy`:
+
+[source,js]
+--------------------------------------------------
+POST /twitter/_update_by_query
+{
+  "query": { <1>
+    "term": {
+      "user": "kimchy"
+    }
+  }
+}
+--------------------------------------------------
+// AUTOSENSE
+
+<1> The query must be passed as a value to the `query` key, in the same
+way as the {ref}/search-search.html[Search API]. You can also use the `q`
+parameter in the same way as the search api.
+
+So far we've only been touching documents. This is useful for picking up new
+properties if the mapping has `"dynamic": "false"` but its only half the fun.
+`_update_by_query` supports the `script` object to define how to actually
+update the document. This will prefix the `user` field with `other` on all of
+kimchy's tweets:
+[source,js]
+--------------------------------------------------
+POST /twitter/_update_by_query
+{
+  "script": {
+    "inline": "ctx._source.user = \"other\" + ctx._source.user"
+  },
+  "query": {
+    "term": {
+      "user": "kimchy"
+    }
+  }
+}
+--------------------------------------------------
+// AUTOSENSE
+
+Just as in {ref}/docs-update.html[Update API] you can set `ctx.op = "noop"` if
+your script decides that it doesn't have to make any changes. That will cause
+`_update_by_query` to omit that document from its updates. Setting `ctx.op` to
+anything else is an error. Setting any other field in `ctx` is an error.
+
+This API doesn't allow you to move the documents it touches, just modify their
+source. This is intentional! We've made no provisions for removing the document
+from its original location.
+
+It's also possible to do this whole thing on multiple indexes and multiple
+types at once, just like the search API:
+
+[source,js]
+--------------------------------------------------
+POST /twitter,blog/tweet,post/_update_by_query
+--------------------------------------------------
+// AUTOSENSE
+
+And `routing` works, on the off chance that you need it:
```

Comment:
This confused me a bit. If we provide routing, what does that mean? Which routing value is then used for indexing??? (explicitly state that it is the original value)